contribute.md

```diff
@@ -3,12 +3,251 @@ layout: page
+  prefix like  `network_module` for the `network` role. While they are usually 
+  implementation details and not intended for direct use in playbooks, the unfortunate 
+  fact is that importing a role makes them available to the rest of the playbook and therefore
+  creates opportunities for name collisions.
+
+* Moreover, internal variables (those that are not expected to be set by users) are to
+  be prefixed by two underscores: `__network_variable`. This includes variables set by
+  set_fact and register, because they persist in the namespace after the role has
+  finished!
+
+* Do not use special characters other than underscore in variable names, even if
+  YAML/JSON allow them. (Using such variables in Jinja2 or Python would be then very
+  confusing and probably not functional.)
+
+*Find more explanation on this matter in the [meta
+standards](https://github.com/oasis-roles/meta_standards#naming-things).*
+
+
+### Write a good commit message
+Here are a few rules to keep in mind while writing a commit message
+
+   1. Separate subject from body with a blank line
+   2. Limit the subject line to 50 characters
+   3. Capitalize the subject line
+   4. Do not end the subject line with a period
+   5. Use the imperative mood in the subject line
+   6. Wrap the body at 72 characters
+   7. Use the body to explain what and why vs. how
+
+ A good commit message looks something like this
+```
+  Summarize changes in around 50 characters or less
+
+ More detailed explanatory text, if necessary. Wrap it to about 72
+ characters or so. In some contexts, the first line is treated as the
+ subject of the commit and the rest of the text as the body. The
+ blank line separating the summary from the body is critical (unless
+ you omit the body entirely); various tools like `log`, `shortlog`
+ and `rebase` can get confused if you run the two together.
+
+ Explain the problem that this commit is solving. Focus on why you
+ are making this change as opposed to how (the code explains that).
+ Are there side effects or other unintuitive consequences of this
+ change? Here's the place to explain them.
+
+ Further paragraphs come after blank lines.
+
+  - Bullet points are okay, too
+
+  - Typically a hyphen or asterisk is used for the bullet, preceded
+    by a single space, with blank lines in between, but conventions
+    vary here
+
+ If you use an issue tracker, put references to them at the bottom,
+ like this:
+
+ Resolves: #123
+ See also: #456, #789
+
+Do not forget to sign your commit! Use `git commit -s`
+
+```
+
+This is taken from [chris beams git commit](https://chris.beams.io/posts/git-commit/).
+You may want to read this for a more detailed explanation (and links to other posts on
+how to write a good commit message). This content is licensed under 
+[CC-BY-SA](https://creativecommons.org/licenses/by-sa/4.0/).
+
+### Debugging
+
+When using the `nm` provider, NetworkManager create a checkpoint and reverts the changes
```

Comment:
this is network role specific

If you want to say something about debugging for all roles, you should mention using `TEST_DEBUG=true`:
```
TEST_DEBUG=true TEST_SUBJECTS=CentOS-8-GenericCloud-8.1.1911-20200113.3.x86_64.qcow2 \
ANSIBLE_STDOUT_CALLBACK=debug \
ansible-playbook -vv -i /usr/share/ansible/inventory/standard-inventory-qcow2 \
tests/tests_default.yml
```
Using `ANSIBLE_STDOUT_CALLBACK=debug` will format the output in a much more human friendly manner, and using `TEST_DEBUG=true` will allow you to `ssh` into the managed host after the test has run.  It will print out instructions about the exact command to use to `ssh`, and how to destroy the managed host after you are done.